README.md

```diff
@@ -20,24 +20,92 @@ Specification for binding services to runtime applications running in Kubernetes
 
 Main section of the doc.  Has sub-section that outline the design.
 
-### 1.  Making a service bindable
+### Making a service bindable
 
-*  What metadata is needed for a service to be bindable (e.g. name of fields it supports, name of secret it produces, etc)
-*  Where is this information located
+#### Recommended
+For a service to be bindable it **SHOULD** provide:
```

Comment:
Could we improve/organize this section to include different 'involved' levels of providing binding information?  I would be uncomfortable if what the spec recommends has a very high bar for entry.

Do we really want to call out anything as recommended? I would rather suggest we point out advantages/disadvantages.

Reply:
It might be difficult to target interoperability if we don't at least prescribe some recommendations…

Reply:
I think the minimum bar that being proposed (i.e. simply annotating the relevant `status` fields to map to the spec's schema) is pretty reasonable, requiring no code changes from the Operator if they don't want to - just CSV or CRD augmenting.  

We could add a section in the spec that provides a "pre-spec compliance" path, where the Service Binding Operator can help as that bridge that maps the existing fields to spec compliant fields.  But even that sometimes is not easy for end-users.  Take the `Strimzi` Operator for example.  It has this in its status:

```
  listeners:
- addresses:
 - host: my-cluster-kafka-bootstrap.kafka.svc
   port: 9092
 type: plain
```

And to make that work with the Service Binding Operator this is the syntax @navidsh had to come up with:

```
  customEnvVar:
    - name: MP_MESSAGING_CONNECTOR_LIBERTY_KAFKA_BOOTSTRAP_SERVERS
      value: |-
        {{- range .status.listeners -}}
          {{- if and (eq .type "plain") (gt (len .addresses) 0) -}}
            {{- with (index .addresses 0) -}}
              {{ .host }}:{{ .port }}
            {{- end -}}
          {{- end -}}
        {{- end -}}
```

All that could be made automatic (and spec compliant) if Strimzi simply annotates its CSV statusDescriptors with the `x-descriptors` proposed here.

Reply:
As for the syntax for this `x-descriptors`, I think it'd be good to support both Go templates and also JSONPath.
JSONPath is already used in `kubectl`/`oc` and I personally think it's more well-known comparing to writing Go Templates. As of now, `customEnvVar` in Service Binding Request only allows Go Templates syntax.

Reply:
thanks for the great comments guys!  I have updated the PR and added 2 new comments to outline the new sections.